shell/platform/android/android_shell_holder.cc

```
@@ -142,12 +168,79 @@ const flutter::Settings& AndroidShellHolder::GetSettings() const {
+      [&jni_facade, android_context, &weak_platform_view](Shell& shell) {
+        std::unique_ptr<PlatformViewAndroid> platform_view_android;
+        platform_view_android = std::make_unique<PlatformViewAndroid>(
+            shell,                   // delegate
+            shell.GetTaskRunners(),  // task runners
+            jni_facade,              // JNI interop
+            android_context          // Android context
+        );
+        weak_platform_view = platform_view_android->GetWeakPtr();
+        shell.OnDisplayUpdates(DisplayUpdateType::kStartup,
+                               {Display(jni_facade->GetDisplayRefreshRate())});
+        return platform_view_android;
+      };
+
+  Shell::CreateCallback<Rasterizer> on_create_rasterizer = [](Shell& shell) {
+    return std::make_unique<Rasterizer>(shell);
+  };
+
+  // TODO(xster): could be worth tracing this to investigate whether
+  // the IsolateConfiguration could be cached somewhere.
+  auto config = BuildRunConfiguration(asset_manager_, entrypoint, libraryUrl);
+  if (!config) {
+    // If the RunConfiguration was null, the kernel blob wasn't readable.
+    // Fail the whole thing.
+    return nullptr;
+  }
+
+  std::unique_ptr<flutter::Shell> shell = shell_->Spawn(
+      std::move(config.value()), on_create_platform_view, on_create_rasterizer);
+
+  return std::make_unique<AndroidShellHolder>(GetSettings(), jni_facade,
+                                              thread_host_, std::move(shell),
+                                              weak_platform_view);
+}
+
+void AndroidShellHolder::Launch(std::shared_ptr<AssetManager> asset_manager,
+                                std::string entrypoint,
+                                std::string libraryUrl) {
   if (!IsValid()) {
     return;
   }
 
-  shell_->RunEngine(std::move(config));
+  asset_manager_ = asset_manager;
+  auto config = BuildRunConfiguration(asset_manager, entrypoint, libraryUrl);
+  if (!config) {
+    return;
```

FML_LOG instead of failing silently?

---

It does on line 280 which is the only way to end up here.